FIAT/tensor_product.py

```diff
@@ -240,16 +267,43 @@ def space_dimension(self):
         # number of dofs just multiplies
         return self.A.space_dimension() * self.B.space_dimension()
 
-    def tabulate(self, order, points):
+    def tabulate(self, order, points, entity=None):
         """Return tabulated values of derivatives up to given order of
         basis functions at given points."""
 
-        Asdim = self.A.get_reference_element().get_spatial_dimension()
-        Bsdim = self.B.get_reference_element().get_spatial_dimension()
-        pointsA = [point[0:Asdim] for point in points]
-        pointsB = [point[Asdim:Asdim + Bsdim] for point in points]
-        Atab = self.A.tabulate(order, pointsA)
-        Btab = self.B.tabulate(order, pointsB)
+        # Factor the entity argument to get entities of the component elements
+        if entity is None or entity[0] == 2:
+            factored_entity = ((self.A.ref_el.get_spatial_dimension(), 0),
+                               (self.B.ref_el.get_spatial_dimension(), 0))
+        else:
+            (dimA, dimB), (idA, idB) = entity
+            factored_entity = (dimA, idA), (dimB, idB)
+        entityA, entityB = factored_entity
```

Comment:
No need to assign to `factored_entity` first, and then split it. You could just assign `entityA, entityB = ...` directly.